backend/app.py

```
@@ -913,6 +914,27 @@ def updateUser(*, context, user_id):
         )
 
 
+#Update User data info
+
+@app.route("/api/user/<user_id>/data", methods=["PATCH"])
+def patchUserData(user_id):
+    """
+    Endpoint to update the 'name' and 'email' fields of a user's 'data'
+    """
+    try:
+        patch_data = request.get_json()
+
+        if patch_data is None or not isinstance(patch_data, dict):
+            return jsonify({"error": "Invalid or missing JSON payload"}), 400
+
```

@egdagger It would be great to validate that the user_id in the request is valid before proceeding. For instance, if we expect an integer, we should try casting it to int and handle any ValueError by returning a 400 Bad Request. We can also confirm whether the user actually exists in our database (e.g., get_user_by_id) and return a 404 if not found. This way, we fail early for invalid user IDs or non-existent users, improving robustness and clarity in our API responses.

---

The user existence in our DB is checked in the function patch_user_data in the cosmo_db.py file (line 334)